tasks/eva_3090/archive_to_lts.py

```diff
@@ -0,0 +1,98 @@
+import gzip
+import shutil
+import tarfile
+import os.path
+from argparse import ArgumentParser
+
+from ebi_eva_common_pyutils.logger import logging_config
+from retry import retry
+
+logging_config.add_stdout_handler()
+logger = logging_config.get_logger(__name__)
+
+
+def make_tarfile(output_filename, source_dir):
+    with tarfile.open(output_filename, "w") as tar:
+        tar.add(source_dir, arcname=os.path.basename(source_dir))
+
+
+def is_compressed(file_path):
+    compressed_ext = ['.gz', '.zip', '.bz', '.tbi', '.csi']
+    for ext in compressed_ext:
+        if file_path.endswith(ext):
+            return True
+    return False
+
+
+def compress(src_file_path, dest_file_path):
+    MEG = 2 ** 20
+    with open(src_file_path, 'rb') as f_in:
+        with gzip.open(dest_file_path, 'wb') as f_out:
+            shutil.copyfileobj(f_in, f_out, length=16 * MEG)
+
+
+def matches(name, patterns):
+    return any((pattern for pattern in patterns if pattern in name))
+
+
+@retry(tries=5, delay=3, backoff=2, logger=logger)
+def retriable_remove(path):
+    shutil.rmtree(path)
+
+
+def archive_directory(root_dir, scratch_dir, destination_dir, filter_patterns=None):
+
+    root_dir_name = os.path.basename(root_dir)
+    logger.info(f'Archive {root_dir_name} from {root_dir}')
+    parent_root_dir = os.path.dirname(root_dir)
+    for base, dirs, files in os.walk(root_dir, topdown=True, followlinks=False):
+        filtered_dir = []
+        for d in dirs:
+            if matches(d, filter_patterns):
+                logger.info(f'Ignore directory {d} because of filters: {filter_patterns}')
+            else:
+                filtered_dir.append(d)
+        # modify dirs in place
+        dirs[:] = filtered_dir
+        src_basename = os.path.relpath(base, parent_root_dir)
+        scratch_dest_dir = os.path.join(scratch_dir, src_basename)
+        if matches(src_basename, filter_patterns):
+
+            continue
+        os.makedirs(scratch_dest_dir, exist_ok=True)
+        for fname in files:
+            src_file_path = os.path.join(base, fname)
+            dest_file_path = os.path.join(scratch_dest_dir, fname)
```

Comment:
Guessing we are using hps as the intermediate destination for scratch here? It could significantly speed up the I/O involved...

Reply:
This is how it was done practically yes.